### PR TITLE
LITE-23422 improve protocol v2

### PR DESCRIPTION
**Pull request by ffaraoneim**

### CODE REVIEW

**Review by marcserrat — APPROVED** [review]

**Review by gab832 — APPROVED** [review]
Good job! I like the renaming!

**Comment by Sainomori** [comment]
As soon as we renamed `transform_data_to_legacy` to `transform_data_to_v1` maybe it's a good idea to rename `SETUP_REQUEST_DATA_LEGACY` with `SETUP_REQUEST_DATA_V1` in tests.

**Review by Sainomori — APPROVED** [review]

File: connect/eaas/core/dataclasses.py [hunk 1]
```diff
@@ -56,32 +56,31 @@ class MessageType:
...
-    countdown: int = 0
-    runtime: float = 0.0
-    output: Optional[str]
     correlation_id: Optional[str]
     reply_to: Optional[str]
 
 
+class TaskOutput(BaseModel):
+    result: str
+    data: Optional[Any]
```
Comment (bdjilka) [inline connect/eaas/core/dataclasses.py:65]: `data` field is for previous `output`?
Moreover, if EaaS we use slicing for it, so `Optional[Any]` can potentially cause errors.

Comment (bdjilka) [reply connect/eaas/core/dataclasses.py:65]: Saw, that `error` is for `output`. For what than data is needed?

Comment (ffaraoneim) [reply connect/eaas/core/dataclasses.py:65]: No, data is for interactive tasks, in general it is a dictionary but can be any additional output data.

**Review by bdjilka — APPROVED** [review]

**Pull request merged**